File: Demo/MTA/FirstViewController.m
//
//  FirstViewController.m
//  MTA-Demo
//
//  Created by WQY on 12-12-10.
//  Copyright (c) 2012年 developer. All rights reserved.
//

#import "FirstViewController.h"
#import "MTA.h"
#import "MTAConfig.h"

@interface FirstViewController ()
@end

@implementation FirstViewController

- (IBAction)clickNormaltButton:(id)sender {
	[MTA trackCustomKeyValueEvent:@"NormalEvent" props:@{ @"para1": @"value1" }];

	UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Event"
													message:@"生成普通事件"
												   delegate:nil
										  cancelButtonTitle:@"OK"
										  otherButtonTitles:nil];

	[alert show];
}

- (IBAction)clickErrorButton:(id)sender {
	[MTA trackError:@"I\'m an unexpected error"];
	UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Event"
													message:@"生成Error事件"
												   delegate:nil
										  cancelButtonTitle:@"OK"
										  otherButtonTitles:nil];
	[alert show];
}

- (IBAction)clickStartButton:(id)sender {
	[MTA trackCustomKeyValueEventBegin:@"TimeEvent" props:nil];

	UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Event"
													message:@"开始计时事件"
												   delegate:nil
										  cancelButtonTitle:@"OK"
										  otherButtonTitles:nil];
	[alert show];
}
- (IBAction)clickEndButton:(id)sender {
	[MTA trackCustomKeyValueEventEnd:@"TimeEvent" props:nil];
	UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Event"
													message:@"结束计时事件"
												   delegate:nil
										  cancelButtonTitle:@"OK"
										  otherButtonTitles:nil];
	[alert show];
}

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
	self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
	if (self) {
		self.title = NSLocalizedString(@"页面1", @"页面1");
		self.tabBarItem.image = [UIImage imageNamed:@"first"];
	}
	return self;
}

@end

File: MTA-Swift-Demo/FirstViewController.swift
//
//  FirstViewController.swift
//  MTA-Swift-Demo
//
//  Created by tyzual on 2018/6/20.
//  Copyright © 2018 developer. All rights reserved.
//

import UIKit

class FirstViewController: UIViewController {

	required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
		self.title = NSLocalizedString("页面1", comment: "页面1")
		self.tabBarItem.image = UIImage(named: "first")
	}
	
	override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
		super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
		self.title = NSLocalizedString("页面1", comment: "页面1")
		self.tabBarItem.image = UIImage(named: "first")
	}
	
    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
	
	@IBAction func clickNormalButton(_ sender: Any) {
		MTA.trackCustomKeyValueEvent("NormalEvent", props: ["para1":"value1"])
		
		UIAlertView(title: "Event", message: "普通事件", delegate: nil, cancelButtonTitle: "OK").show()
	}
	
	@IBAction func clickErrorButton(_ sender: Any) {
		MTA.trackError("I\'m an unexpected error")
		
		UIAlertView(title: "Event", message: "Error件", delegate: nil, cancelButtonTitle: "OK").show()
	}
	
	@IBAction func clickStartButton(_ sender: Any) {
		MTA.trackCustomKeyValueEventBegin("TimeEvent", props: nil)
		
		UIAlertView(title: "Event", message: "开始计时事件", delegate: nil, cancelButtonTitle: "OK").show()
	}
	
	@IBAction func clickEndButton(_ sender: Any) {
		MTA.trackCustomKeyValueEventEnd("TimeEvent", props: nil)
		
		UIAlertView(title: "Event", message: "结束计时事件", delegate: nil, cancelButtonTitle: "OK").show()
	}
}
